tests/mock_crust/utils.rs

```diff
@@ -295,20 +297,17 @@ pub fn poll_all(nodes: &mut [TestNode], clients: &mut [TestClient]) -> bool {
 /// Polls and processes all events, until there are no unacknowledged messages left and clearing
 /// the nodes' state triggers no new events anymore.
 pub fn poll_and_resend(nodes: &mut [TestNode], clients: &mut [TestClient]) {
+    let mut clock_advance_count = 0;
     for _ in 0..MAX_POLL_CALLS {
         if poll_all(nodes, clients) {
-            let mut call_count = 1;
-            while resend_unacknowledged(nodes, clients) && poll_all(nodes, clients) {
-                call_count += 1;
-                assert_ne!(call_count,
-                           MAX_POLL_CALLS,
-                           "Polling and resending unacknowledged has been called {} times.",
-                           MAX_POLL_CALLS);
-            }
-            nodes.iter_mut().foreach(|node| node.inner.clear_state());
-        } else {
+            clock_advance_count = 0;
+        } else if clock_advance_count > 3 {
+            // Expect advance at least three times, given NODE_CONNECT_TIMEOUT_SECS is 60s and
+            // ACK_TIMEOUT_SECS is 20s
             return;
         }
+        FakeClock::advance_time(20 * 1000 + 1);
```

Comment:
I think I'd be in favour of taking the hit of more conditionally-compiled code to expose `ACK_TIMEOUT_SECS` and `NODE_CONNECT_TIMEOUT_SECS` rather than using the hard coded values of `20` and `3` here.  Same for https://github.com/maidsafe/routing/pull/1438/files#r113455846